de.metas.banking/de.metas.banking.base/src/test/java/de/metas/banking/service/impl/BankStatementPaymentBLTest.java

```diff
@@ -0,0 +1,339 @@
+import de.metas.banking.model.I_C_BankStatementLine;
+import de.metas.banking.model.I_C_Payment;
+import de.metas.banking.model.validator.C_BankStatement;
+import de.metas.banking.model.validator.C_BankStatementLine;
+import de.metas.banking.model.validator.C_BankStatementLine_Ref;
+import de.metas.banking.payment.impl.BankStatmentPaymentBL;
+import de.metas.bpartner.BPartnerId;
+import de.metas.business.BusinessTestHelper;
+import de.metas.document.engine.DocStatus;
+import de.metas.money.CurrencyId;
+import de.metas.organization.OrgId;
+import de.metas.payment.TenderType;
+import de.metas.payment.api.IPaymentBL;
+import de.metas.util.Services;
+import de.metas.util.time.SystemTime;
+import org.adempiere.ad.modelvalidator.IModelInterceptorRegistry;
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.adempiere.test.AdempiereTestHelper;
+import org.compiere.model.I_C_BP_BankAccount;
+import org.compiere.model.I_C_BPartner;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import java.math.BigDecimal;
+import java.sql.Timestamp;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class BankStatementPaymentBLTest
+{
+	private final Timestamp statementDate = SystemTime.asTimestamp();
+	private final String metasfreshIban = "123456";
+
+	private final Timestamp valutaDate = SystemTime.asTimestamp();
+
+	@BeforeEach
+	void setUp()
+	{
+		AdempiereTestHelper.get().init();
+
+		final IModelInterceptorRegistry modelInterceptorRegistry = Services.get(IModelInterceptorRegistry.class);
+		modelInterceptorRegistry.addModelInterceptor(C_BankStatementLine_Ref.instance);
+		modelInterceptorRegistry.addModelInterceptor(C_BankStatement.instance);
+		modelInterceptorRegistry.addModelInterceptor(C_BankStatementLine.instance);
+	}
+
+	private void paymentChecks(final BigDecimal expectedPayAmt, final int expectedC_payment_id, final boolean expectedIsReceipt, final int expectedC_BP_BankAccount_ID)
+	{
+		final I_C_Payment payment = InterfaceWrapperHelper.load(expectedC_payment_id, I_C_Payment.class);
+		assertNotNull(payment);
+		assertEquals(expectedPayAmt, payment.getPayAmt());
+		assertTrue(payment.isReconciled());
+		assertEquals(expectedIsReceipt, payment.isReceipt());
+		assertEquals(DocStatus.Completed, DocStatus.ofCode(payment.getDocStatus()));
+		assertEquals(expectedC_BP_BankAccount_ID, payment.getC_BP_BankAccount_ID());
+
+		// can't test `payment.getC_DocType_ID()` as it is set by `PaymentsForInvoicesCreator`, and during test there's no DocTypes
+	}
+
+	@Nested
+	class ExistingPayments
+	{
+		@Test
+		void vendorOneMatchingPaymentExists_DifferentInvoiceOnBSL()
+		{
+			// TODO tbp: check with mark in a followup task about this usecase.
```

Comment:
what's the status here.
basically implement it or delete it.

Reply:
Since there will be a followup task, i will do neither now. After we decide how we do the matching of invoices, i will implement this test.

If i will implement it now, it will always fail until we complete the followup task, and this is something i don't approve of.